ansible/roles-infra/infra-osp-template-create/tasks/create_osp_template.yml

```diff
@@ -1,69 +1,98 @@
-          command: >-
-            openstack stack update -f json --wait
-            -t {{ heat_master_template }}
-            base-stack-{{ guid }}
-          register: r_hot_out_update
-
-    - when: r_show.rc != 0
-      block:
+  - name: Update Heat stack from generated template
+    when: r_stack_list.stdout | from_json | json_query(stack_name)
+    command: >-
+      openstack stack update -f json --wait
+      -t {{ heat_master_template }}
+      base-stack-{{ guid }}
+    vars:
+      stack_name: >-
+        [?"Stack Name"=='base-stack-{{ guid }}']
+    register: r_hot_out_update
 
-        - name: Create Heat stack from generated template
-          command: >-
-            openstack stack create -f json --wait
-            -t {{ heat_master_template }}
-            base-stack-{{ guid }}
-          register: r_hot_out_create
+  - name: Create Heat stack from generated template
+    when: not r_stack_list.stdout | from_json | json_query(stack_name)
+    command: >-
+      openstack stack create -f json --wait
+      -t {{ heat_master_template }}
+      base-stack-{{ guid }}
+    vars:
+      stack_name: >-
+        [?"Stack Name"=='base-stack-{{ guid }}']
+    register: r_hot_out_create
 
-    - name: Grab outputs
-      command: >-
-        openstack stack show -f json base-stack-{{ guid }}
-      changed_when: false
-      register: r_hot_outputs
+  - name: Grab outputs
+    command: >-
+      openstack stack show -f json base-stack-{{ guid }}
+    changed_when: false
+    register: r_hot_outputs
 
-    - debug:
-        var: r_hot_outputs
-        verbosity: 3
+  - debug:
+      var: r_hot_outputs
+      verbosity: 3
 
-    - set_fact:
-        hot_outputs: "{{ (r_hot_outputs.stdout | from_json) }}"
-    
-    - debug:
-        var: hot_outputs
-        verbosity: 3
+  - set_fact:
+      hot_outputs: "{{ (r_hot_outputs.stdout | from_json) }}"
+  
+  - debug:
+      var: hot_outputs
+      verbosity: 3
 
-    - name: Save infra_key content
-      set_fact:
-        infra_private_key_content: "{{ hot_outputs | json_query(query) }}"
-      vars:
-        query: "outputs[?@.output_key=='{{ guid }}-infra_key'].output_value|[0]"
+  - name: Save infra_key content
+    set_fact:
+      infra_private_key_content: "{{ hot_outputs | json_query(query) }}"
+    vars:
+      query: "outputs[?@.output_key=='{{ guid }}-infra_key'].output_value|[0]"
 
-    - name: Set infra_ssh_key path
-      set_fact:
-        infra_ssh_key: "{{ output_dir }}/{{ guid }}_infra_ssh_key.pem"
+  - name: Set infra_ssh_key path
+    set_fact:
+      infra_ssh_key: "{{ output_dir }}/{{ guid }}_infra_ssh_key.pem"
 
-    - name: Copy infra_key content to output_dir
-      copy:
-        dest: "{{ infra_ssh_key }}"
-        content: "{{ infra_private_key_content }}"
-        mode: 0600
+  - name: Copy infra_key content to output_dir
+    copy:
+      dest: "{{ infra_ssh_key }}"
+      content: "{{ infra_private_key_content }}"
+      mode: 0600
```

Comment:
As long as we are touching this line, can we change to use symbolic style, `u=rw,g=,o=`?
Symbolic is best practice as it is clearer for people reading the code and is less likely to to run into octal/decimal issues.

Reply:
This line was only touched to fix indentation with the previous block removal. Changing this, even though small, is unnecessary and is outside of the scope of this PR so I am going to leave it for now.

Reply:
Note `openstack stack list --property stack_status=CREATE_IN_PROGRESS --property stack_status=UPDATE_IN_PROGRESS` will NOT work.  It will show everything.

Reply:
Moving this to its own comment since it is unrelated to what JK mentioned here.